tests/unit/amo/components/TestFlagReviewMenu.js

```diff
@@ -0,0 +1,161 @@
+      expect(authButton).toHaveProp('location', location);
+    });
+
+    it('does not let you flag your own review', () => {
+      const review = denormalizeReview({ ...fakeReview });
+      dispatchSignInActions({ store, userId: review.userId });
+      const { menu } = renderMenu({ review });
+
+      const items = menu.find(ListItem);
+      expect(items).toHaveLength(1);
+      expect(items.at(0).html()).toContain(
+        'You cannot flag your own review'
+      );
+    });
+
+    it('does not let you flag your own response', () => {
+      const review = denormalizeReview({ ...fakeReview });
+      dispatchSignInActions({ store, userId: review.userId });
+      const { menu } = renderMenu({ review, isDeveloperReply: true });
+
+      const items = menu.find(ListItem);
+      expect(items).toHaveLength(1);
+      expect(items.at(0).html()).toContain(
+        'You cannot flag your own response'
+      );
+    });
+  });
+
+  describe('prompts and FlagReview configuration', () => {
+    it('provides a tooltip to flag reviews', () => {
+      const root = render();
+
+      expect(root.find(TooltipMenu))
+        .toHaveProp('openerTitle', 'Flag this review');
+    });
+
+    it('provides a tooltip to flag developer responses', () => {
+      const root = render({ isDeveloperReply: true });
+
+      expect(root.find(TooltipMenu))
+        .toHaveProp('openerTitle', 'Flag this developer response');
+    });
+
+    it('configures FlagReview to flag as spam', () => {
+      const review = denormalizeReview(fakeReview);
+      const { menu } = renderMenu({ review });
+
+      const flag = menu.find('.FlagReviewMenu-flag-spam-item')
+        .find(FlagReview);
+      expect(flag).toHaveProp('review', review);
+      expect(flag).toHaveProp('reason', REVIEW_FLAG_REASON_SPAM);
+      expect(flag).toHaveProp('wasFlaggedText');
+      expect(flag).toHaveProp('buttonText');
+    });
+
+    it('configures FlagReview to flag for language', () => {
+      const review = denormalizeReview(fakeReview);
+      const { menu } = renderMenu({ review });
+
+      const flag = menu.find('.FlagReviewMenu-flag-language-item')
+        .find(FlagReview);
+      expect(flag).toHaveProp('review', review);
+      expect(flag).toHaveProp('reason', REVIEW_FLAG_REASON_LANGUAGE);
+      expect(flag).toHaveProp('wasFlaggedText');
+      expect(flag).toHaveProp('buttonText');
+    });
+
+    it('configures FlagReview to flag as bug/support', () => {
+      const review = denormalizeReview(fakeReview);
+      const { menu } = renderMenu({ review });
+
+      const flag = menu.find('.FlagReviewMenu-flag-bug-support-item')
+        .find(FlagReview);
+      expect(flag).toHaveProp('review', review);
+      expect(flag).toHaveProp('reason', REVIEW_FLAG_REASON_BUG_SUPPORT);
+      expect(flag).toHaveProp('wasFlaggedText');
+      expect(flag).toHaveProp('buttonText');
+    });
+
+    it('does not prompt you to flag a response as a bug/support', () => {
```

Comment:
I get why this is but it just took me a few seconds to parse, understand the intention here, and then realise why. Feel free to disagree but I think a comment explaining why "responses can't be flagged as bug reports" might be handy. 😄

Reply:
ok, I added a comment to `FlagReviewMenu` about it.